gunicorn/app/wsgiapp.py

```diff
@@ -31,11 +31,15 @@ def init(self, parser, opts, args):
             from .pasterapp import paste_config
             return paste_config(self.cfg, self.cfgurl, self.relpath)
 
-        if len(args) < 1:
+        if len(args) > 0:
+            self.cfg.set("default_proc_name", args[0])
```

Comment:
since it is always using `self.app_uri` it should be set outside of the `if` imo so it's edited on one place.